pkg/scanners/scannerv4/vm_convert.go

```diff
@@ -0,0 +1,166 @@
+package scannerv4
+
+import (
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/generated/storage"
+	pkgCVSS "github.com/stackrox/rox/pkg/cvss"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+// ToVirtualMachineScan converts a scan report to the format needed to enrich a virtual machine with scan data.
+func ToVirtualMachineScan(r *v4.VulnerabilityReport) *storage.VirtualMachineScan {
+	return &storage.VirtualMachineScan{
+		ScanTime: protocompat.TimestampNow(),
+		// TODO: find an actual operating system source
+		OperatingSystem: "",
+		Notes:           toVirtualMachineScanNotes(r.GetNotes()),
+		Components:      toVirtualMachineComponents(r),
+	}
+}
+
+func toVirtualMachineScanNotes(notes []v4.VulnerabilityReport_Note) []storage.VirtualMachineScan_Note {
+	result := make([]storage.VirtualMachineScan_Note, 0, len(notes))
+	for _, note := range notes {
+		switch note {
+		case v4.VulnerabilityReport_NOTE_UNSPECIFIED:
+			result = append(result, storage.VirtualMachineScan_UNSET)
+		case v4.VulnerabilityReport_NOTE_OS_UNKNOWN:
+			result = append(result, storage.VirtualMachineScan_OS_UNKNOWN)
+		case v4.VulnerabilityReport_NOTE_OS_UNSUPPORTED:
+			result = append(result, storage.VirtualMachineScan_OS_UNSUPPORTED)
+		default:
+			result = append(result, storage.VirtualMachineScan_UNSET)
+		}
+	}
+	return result
+}
+
+func toVirtualMachineComponents(r *v4.VulnerabilityReport) []*storage.EmbeddedVirtualMachineScanComponent {
+	packages := r.GetContents().GetPackages()
+	result := make([]*storage.EmbeddedVirtualMachineScanComponent, 0, len(packages))
+	for _, pkg := range packages {
+		vulnerabilityIDs := r.GetPackageVulnerabilities()[pkg.GetId()].GetValues()
+		vulnerabilitiesByID := r.GetVulnerabilities()
```

Comment:
Need to check what is returned in those two fields as I don't remember precisely. Have you checked an exemplary Vulnerability scan returned from Scanner?